monai/fl/client/monai_algo.py

```diff
@@ -80,55 +68,38 @@ def compute_weight_diff(global_weights, local_var_dict):
     return weight_diff
 
 
-def check_bundle_config(parser):
-    for k in RequiredBundleKeys:
-        if parser.get(k, None) is None:
-            raise KeyError(f"Bundle config misses required key `{k}`")
-
-
-def disable_ckpt_loaders(parser):
-    if BundleKeys.VALIDATE_HANDLERS in parser:
-        for h in parser[BundleKeys.VALIDATE_HANDLERS]:
-            if ConfigComponent.is_instantiable(h):
-                if "CheckpointLoader" in h["_target_"]:
-                    h["_disabled_"] = True
-
-
 class MonaiAlgoStats(ClientAlgoStats):
     """
     Implementation of ``ClientAlgoStats`` to allow federated learning with MONAI bundle configurations.
 
     Args:
-        bundle_root: path of bundle.
-        config_train_filename: bundle training config path relative to bundle_root. Can be a list of files;
-            defaults to "configs/train.json".
+        workflow: the bundle workflow to execute, usually it's training, evaluation or inference.
         config_filters_filename: filter configuration file. Can be a list of files; defaults to `None`.
+        data_stats_transform_list: transforms to apply for the data stats result.
         histogram_only: whether to only compute histograms. Defaults to False.
     """
 
     def __init__(
         self,
-        bundle_root: str,
-        config_train_filename: str | list | None = "configs/train.json",
+        workflow: BundleWorkflow,
         config_filters_filename: str | list | None = None,
-        train_data_key: str | None = BundleKeys.TRAIN_DATA,
-        eval_data_key: str | None = BundleKeys.VALID_DATA,
         data_stats_transform_list: list | None = None,
         histogram_only: bool = False,
     ):
         self.logger = logger
-        self.bundle_root = bundle_root
-        self.config_train_filename = config_train_filename
+        if not isinstance(workflow, BundleWorkflow):
+            raise ValueError("workflow must be a subclass of BundleWorkflow.")
+        if workflow.get_workflow_type() is None:
+            raise ValueError("workflow doesn't specify the type.")
+        self.workflow = workflow
         self.config_filters_filename = config_filters_filename
-        self.train_data_key = train_data_key
-        self.eval_data_key = eval_data_key
+        self.train_data_key = "train"
```

Comment:
Why is this hard-coded now? Can it not be different for different bundles anymore?

Reply:
I think this key actually only affects the data analysis report result:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/fl/client/monai_algo.py#L261
So I hard-code it to "train" directly, do you have any use-case to set different keys? Or maybe I misunderstand the logic?

Thanks.

Reply:
You are right. These keys are just used internally, and we can keep them constant.

Reply:
Thanks for your confirmation.